ruby/src/IceRuby/Util.h

```diff
@@ -494,14 +494,7 @@ ice_start:
 
 #define ICE_RUBY_CATCH                                                                                                 \
     catch (const ::IceRuby::RubyException& ex) { ICE_RUBY_RETHROW(ex.ex); }                                            \
-    catch (const ::Ice::LocalException&) { ICE_RUBY_RETHROW(convertLocalException(current_exception())); }             \
-    catch (const ::Ice::Exception& ex)                                                                                 \
-    {                                                                                                                  \
-        string msg_ = "unknown Ice exception: " + std::string{ex.ice_id()};                                            \
-        ICE_RUBY_RETHROW(rb_exc_new2(rb_eRuntimeError, msg_.c_str()));                                                 \
-    }                                                                                                                  \
-    catch (const std::bad_alloc& ex) { ICE_RUBY_RETHROW(rb_exc_new2(rb_eNoMemError, ex.what())); }                     \
-    catch (const std::exception& ex) { ICE_RUBY_RETHROW(rb_exc_new2(rb_eRuntimeError, ex.what())); }                   \
-    catch (...) { ICE_RUBY_RETHROW(rb_exc_new2(rb_eRuntimeError, "caught unknown C++ exception")); }
+    catch (const ::std::bad_alloc& ex) { ICE_RUBY_RETHROW(rb_exc_new2(rb_eNoMemError, ex.what())); }                   \
+    catch (...) { ICE_RUBY_RETHROW(convertException(current_exception())); }
```

Comment:
We can simplify ICE_RUBY_TRY/ICE_RUBY_RETHROW further, no need to use separate labels and goto for ice_start/ice_handle_exception

Reply:
Simplified. I don't think my PR helped - we could have simplified earlier.